packages/block/src/header.ts

```diff
@@ -184,9 +184,11 @@ export class BlockHeader {
       throw new Error('invalid header. Less values than expected were received')
     }
 
-    if (opts.common?.isActivatedEIP(1559) && !baseFeePerGas) {
-      // TODO fixme
-      // throw new Error('invalid header. baseFeePerGas should be provided')
+    if (opts.common?.isActivatedEIP(1559) && (baseFeePerGas === undefined)) {
+      const eip1559ActivationBlock = opts.common?.eipBlockBN(1559)?.toArrayLike(Buffer)
```

Comment:
you don't need to add `eipBlockBN` here, you can use hardforkBlockBN like on L275:

https://github.com/ethereumjs/ethereumjs-monorepo/blob/2b58e75bfbb9476fe179ba1bf5b6a5f16dff4bd1/packages/block/src/header.ts#L273-L289

so instead of setting default 7 you want to throw? that would change a lot of internal uses where we simplify to a sensible default on blank init

Reply:
Users have to provide the explicit `initialBaseFee` as `baseFee` in the london transition block in case they use `fromValuesArray` or `fromRLP`. Most users who want to construct a block will use `fromData`. If we would remove this change from the `fromRLP`, then we have this problem that someone can feed us an initial London block which has no baseFee in the RLP, which is invalid, and can then start building upon that wrong chain.

Ah OK regarding the `eipBlockBN`, I see that it is fixed by just getting the London fork. But this is a bit inconsistent, right? If you want a custom chain with 1559 but no other EIPs then `London` fork is at block `null`, it is never activated. I think this is rather inconsistent. It is also weird that we cannot check whenever an EIP is activated in Common, which I think we should add as feature.

Reply:
yeah I can go along with that, if baseFee is not supplied in fromValuesArray then we can error. in that case we probably don't need the initial base fee logic, since if we are building the block we would use fromBlockData and if we are getting it from peers we assume it to be fully correct.

eipBlockBN, yeah it just gets complicated to support that feature, I think we need a refactoring where we set everything up front and don't do custom chain iterations in any get() method (to make the operation quicker). didn't you start on some kind of refactoring like that but it got complicated?